lib/call-cabal-project-to-nix.nix

```diff
@@ -340,6 +340,14 @@ let
     fi
     cp -r ${maybeCleanedSource}/* .
     chmod +w -R .
+    # Remove package.yaml files that have already been converted
+    # to `.cabal` files.  This avoids problems where the `hpack`
+    # version used may be different.
+    for p in $(find . -name package.yaml); do
```

Comment:
Can we merge this with the loop below where we run hpack? so we loop over the package.yaml files and then either run hpack or don't depending on whether the cabal file is there?